clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/agent/TaskHealthCachingAgent.java

```diff
@@ -118,56 +120,144 @@ public TaskHealthCachingAgent(NetflixAmazonCredentials account, String region,
-          taskHealth.setType("loadBalancer");
-          taskHealth.setState(targetHealth);
-          taskHealth.setServiceName(serviceName);
-          taskHealth.setTaskId(task.getTaskId());
-          taskHealth.setTaskArn(task.getTaskArn());
-          taskHealth.setInstanceId(task.getTaskArn());
 
-          taskHealthList.add(taskHealth);
-        }
+      TargetHealthDescription healthDescription = describeTargetHealthResult.getTargetHealthDescriptions().get(0);
+
+      TaskHealth taskHealth = makeTaskHealth(task, serviceName, healthDescription);
+      return taskHealth;
+    }
+    return null;
+  }
+
+  private void evictStaleData(Task task, Service loadBalancerService) {
+    String serviceEvictionKey = Keys.getTaskDefinitionKey(accountName, region, loadBalancerService.getServiceName());
+    serviceEvicitions.add(serviceEvictionKey);
+    String taskEvictionKey = Keys.getTaskKey(accountName, region, task.getTaskId());
+    taskEvicitions.add(taskEvictionKey);
+
+    String taskDefArn = loadBalancerService.getTaskDefinition();
+    String taskDefKey = Keys.getTaskDefinitionKey(accountName, region, taskDefArn);
+    taskDefEvicitions.add(taskDefKey);
+  }
+
+  private TaskHealth makeTaskHealth(Task task, String serviceName, TargetHealthDescription healthDescription) {
+    String targetHealth = healthDescription.getTargetHealth().getState().equals("healthy") ? "Up" : "Unknown";
+
+    TaskHealth taskHealth = new TaskHealth();
+    taskHealth.setType("loadBalancer");
+    taskHealth.setState(targetHealth);
+    taskHealth.setServiceName(serviceName);
+    taskHealth.setTaskId(task.getTaskId());
+    taskHealth.setTaskArn(task.getTaskArn());
+    taskHealth.setInstanceId(task.getTaskArn());
+    return taskHealth;
+  }
+
+  private TaskHealth inferHealthNetworkBindedContainer(AmazonElasticLoadBalancing amazonloadBalancing,
+                                                       Task task,
+                                                       ContainerInstance containerInstance,
+                                                       String serviceName,
+                                                       Service loadBalancerService) {
+    int port = task.getContainers().get(0).getNetworkBindings().get(0).getHostPort();
+
+    List<LoadBalancer> loadBalancers = loadBalancerService.getLoadBalancers();
+
+    for (LoadBalancer loadBalancer : loadBalancers) {
+      if (loadBalancer.getTargetGroupArn() == null || containerInstance.getEc2InstanceId() == null) {
+        continue;
       }
+
+      DescribeTargetHealthResult describeTargetHealthResult;
+      describeTargetHealthResult = amazonloadBalancing.describeTargetHealth(
+        new DescribeTargetHealthRequest().withTargetGroupArn(loadBalancer.getTargetGroupArn()).withTargets(
+          new TargetDescription().withId(containerInstance.getEc2InstanceId()).withPort(port)));
+
+      if (describeTargetHealthResult.getTargetHealthDescriptions().size() == 0) {
```

Comment:
`isEmpty()` -- it's always more efficient than a `size()` operation if you don't actually care about the size.

Reply:
Done